### PR TITLE
refactor: account manager and state handler

### PR DESCRIPTION
**Pull request by aaronc**

### CODE REVIEW

**Comment by tac0turtle** [comment]
can you run fmt to have one ci job green

File: crates/module_system/message_api/src/handler.rs [hunk 1]
```diff
@@ -1,24 +1,53 @@
...
+    }
+
+    /// Handle a system message.
+    fn handle_system(
+        &self,
+        message_packet: &mut MessagePacket,
+        callbacks: &mut dyn HostBackend,
+        allocator: &dyn Allocator,
+    ) -> Result<(), ErrorCode> {
+        Err(ErrorCode::SystemCode(SystemCode::MessageNotHandled))
```
Comment (tac0turtle) [inline crates/module_system/message_api/src/handler.rs:35]: is this a todo or on purpose?

File: crates/module_system/core/src/context.rs [hunk 2]
```diff
@@ -64,22 +86,97 @@ impl<'a> Context<'a> {
...
+    pub fn consume_gas(&self, gas: u64) -> Result<(), ErrorCode> {
+        if self.gas_left.get() < gas {
+            self.gas_left.set(0);
+            return Err(ErrorCode::SystemCode(SystemCode::OutOfGas));
         }
+        self.gas_left.set(self.gas_left.get() - gas);
+        Ok(())
     }
+
+    // /// Dynamically invokes a message.
```
Comment (tac0turtle) [inline crates/module_system/core/src/context.rs]: can we remove the commented out code?

File: crates/module_system/message_api/src/handler.rs (hunk 1 above)
Comment (aaronc) [reply crates/module_system/message_api/src/handler.rs:35]: it's on purpose but probably not necessary

Comment (aaronc) [reply crates/module_system/message_api/src/handler.rs:35]: actually it is used. these are method stubs so implementors don't need to implement them all

File: crates/module_system/core/src/context.rs (hunk 2 above)
Comment (aaronc) [reply crates/module_system/core/src/context.rs]: done

**Review by tac0turtle — APPROVED** [review]

**Pull request merged**